src/main/java/com/nimiq/PeerInfo.java

```diff
@@ -0,0 +1,190 @@
+                if (state.getNumber() == number) {
+                    return state;
+                }
+            }
+            return null;
+        }
+    }
+
+    public enum ConnectionState {
+        NEW(1), CONNECTING(2), CONNECTED(3), NEGOTIATING(4), ESTABLISHED(5), CLOSED(6);
+
+        private int number;
+
+        ConnectionState(int number) {
+            this.number = number;
+        }
+
+        public int getNumber() {
+            return number;
+        }
+
+        @JsonCreator
+        public static ConnectionState fromNumber(int number) {
+            for (ConnectionState state : ConnectionState.values()) {
+                if (state.getNumber() == number) {
+                    return state;
+                }
+            }
+            return null;
+        }
+    }
+
+    private String id;
+    private String address;
+    private AddressState addressState;
+    private ConnectionState connectionState;
+    private int version;
+    private long timeOffset;
+    private String headHash;
+    private long latency;
+    private long rx;
+    private long tx;
+
+    /**
+     * @return Peer id.
+     */
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * @return Peer address. (URL)
+     */
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    /**
+     * @return Address state.
+     */
+    public AddressState getAddressState() {
+        return addressState;
+    }
+
+    public void setAddressState(AddressState addressState) {
+        this.addressState = addressState;
+    }
+
+    /**
+     * @return Connection state.
+     */
+    public ConnectionState getConnectionState() {
+        return connectionState;
+    }
+
+    public void setConnectionState(ConnectionState connectionState) {
+        this.connectionState = connectionState;
+    }
+
+    /**
+     * @return Version.
+     */
+    public int getVersion() {
+        return version;
+    }
+
+    public void setVersion(int version) {
+        this.version = version;
+    }
+
+    /**
+     * @return Time offset. (in milliseconds)
+     */
+    public long getTimeOffset() {
+        return timeOffset;
+    }
+
+    public void setTimeOffset(long timeOffset) {
+        this.timeOffset = timeOffset;
+    }
+
+    /**
+     * @return Hash of the head block the peer is on.
+     */
+    public String getHeadHash() {
+        return headHash;
+    }
+
+    public void setHeadHash(String headHash) {
+        this.headHash = headHash;
+    }
+
+    /**
+     * @return Network latency. (in milliseconds)
+     */
+    public long getLatency() {
+        return latency;
+    }
+
+    public void setLatency(long latency) {
+        this.latency = latency;
+    }
+
+    /**
+     * @return Bytes received.
+     */
+    public long getRx() {
```

Comment:
How about `getBytesReceived()` and `getBytesSent()` below to be completely self-explaining?

Reply:
yes, those names sound better

https://github.com/nimiq-community/java-client/pull/1/commits/647c42ec5dbada8670f2c5695715698f890cc8ff

I actually just kept the original names everywhere
```
> peer "wss://seed1.nimiq-testnet.com:8080/b99034c552e9c0fd34eb95c1cdf17f5e"
Peer b99034c552e9c0fd34eb95c1cdf17f5e:
Address         | wss://seed1.nimiq-testnet.com:8080/b99034c552e9c0fd34eb95c1cdf17f5e
Failed attempts | undefined
A-State         | Established
C-State         | Established
Head hash       | 27a8f74d039ae1372916c4ddf35d3cc10498a2057102a7ed72c1c7f5c4badadb
Time offset     | -149
Latency         | 292
Traffic         | 820 kB RX / 61.5 kB TX
```